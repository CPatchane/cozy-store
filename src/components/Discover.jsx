import React, { Component } from 'react'

import { translate } from 'cozy-ui/react/I18n'

import SmallAppItem from './SmallAppItem'

const DEMO_APPS = [
  {
    slug: 'bank',
    name: 'Bank',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-bank.svg')
  },
  {
    slug: 'sante',
    name: 'Sant√©',
    developer: {name: 'MAIF Ltd'},
    editor: '',
    icon: require('../assets/temp_apps/icon-sante.svg')
  },
  {
    slug: 'mon_logis',
    name: 'Mon Logis',
    developer: {name: 'HoodBrains'},
    editor: '',
    icon: require('../assets/temp_apps/icon-mon_logis.svg')
  },
  {
    slug: 'calendar',
    name: 'Calendar',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-calendar.svg')
  }
]

class Discover extends Component {
  render () {
    const { t } = this.props
    return (
      <div className='sto-discover sto-soon'>
        <h2 className='sto-discover-title'>{t('discover.title')}</h2>
        <div className='sto-discover-get-started'>
          <h3 className='sto-discover-get-started-title'>
            {t('discover.get_started')}
          </h3>
          <div className='sto-discover-get-started-list'>
            {DEMO_APPS.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
              />
            })}
          </div>
        </div>
      </div>
    )
  }
}

export default translate()(Discover)
