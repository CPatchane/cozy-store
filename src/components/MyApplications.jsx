import React, { Component } from 'react'

import { Route } from 'react-router-dom'
import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'

import SmallAppItem from './SmallAppItem'
import ApplicationModal from './ApplicationModal'

class MyApplications extends Component {
  constructor (props) {
    super(props)
    props.fetchApps()
  }

  onAppClick (appSlug) {
    this.props.history.push(`/myapps/${appSlug}/manage`)
  }

  render () {
    const { t, myApps, isFetching, error } = this.props
    return (
      <div className='sto-myapps'>
        <h2 className='sto-myapps-title'>{t('myapps.title')}</h2>
        <div className='sto-myapps-list'>
          {!isFetching && myApps && !!myApps.length &&
            myApps.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
                onClick={() => this.onAppClick(a.slug)}
              />
            })
          }
          {error &&
            <p>{error}</p>
          }
        </div>
        {isFetching &&
          <Spinner
            size='xxlarge'
            loadingType='appsFetching'
            middle='true'
          />
        }

        <Route path='/myapps/:appSlug/manage' render={({ match }) =>
          (<ApplicationModal {...this.props} match={match} />)
        } />
      </div>
    )
  }
}

export default translate()(MyApplications)
