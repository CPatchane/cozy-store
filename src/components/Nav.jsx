import React from 'react'
import { translate } from 'cozy-ui/react/I18n'
import { NavLink } from 'react-router-dom'

export const Nav = ({ t }) => (
  <nav>
    <ul className='coz-nav'>
      <li className='coz-nav-item'>
        <NavLink to='/discover' className='sto-cat-discover coz-nav-link' activeClassName='active'>
          { t('nav.discover') }
        </NavLink>
      </li>
      <li className='coz-nav-item'>
        <NavLink to='/myapps' className='sto-cat-myapps coz-nav-link' activeClassName='active'>
          { t('nav.myapps') }
        </NavLink>
      </li>
    </ul>
  </nav>
)

export default translate()(Nav)
