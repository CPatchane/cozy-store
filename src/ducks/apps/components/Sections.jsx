import React from 'react'
import { translate } from 'cozy-ui/react/I18n'

import AppsSection from './AppsSection'
import { APP_TYPE } from '../index'

const _getSortedByCategories = (appsList) => {
  return appsList.reduce((sortedAppsObject, app) => {
    app.categories.map(c => {
      if (!sortedAppsObject.hasOwnProperty(c)) sortedAppsObject[c] = []
      sortedAppsObject[c].push(app)
    })
    return sortedAppsObject
  }, {})
}

export const Sections = ({t, lang, apps, error, onAppClick}) => {
  if (error) return <p className='u-error'>{error.message}</p>
  const konnectorsCategories = _getSortedByCategories(
    apps.filter(a => a.type === APP_TYPE.KONNECTOR)
  )
  const webAppsCategories = _getSortedByCategories(
    apps.filter(a => a.type === APP_TYPE.WEBAPP)
  )
  return (<div className='sto-sections'>
    {!!Object.keys(webAppsCategories).length &&
      <div className='sto-sections-section'>
        <h2 className='sto-sections-title'>
          {t('sections.applications')}
        </h2>
        {Object.keys(webAppsCategories).map(cat => {
          return (
            <AppsSection
              appsList={webAppsCategories[cat]}
              subtitle={t(`app_categories.${cat}`)}
              onAppClick={onAppClick}
            />
          )
        })}
      </div>
    }
    {!!Object.keys(konnectorsCategories).length &&
      <div className='sto-sections-section'>
        <h2 className='sto-sections-title'>
          {t('sections.konnectors')}
        </h2>
        {Object.keys(konnectorsCategories).map(cat => {
          return (
            <AppsSection
              appsList={konnectorsCategories[cat]}
              subtitle={t(`app_categories.${cat}`)}
              onAppClick={onAppClick}
            />
          )
        })}
      </div>
    }
  </div>)
}

export default translate()(Sections)
