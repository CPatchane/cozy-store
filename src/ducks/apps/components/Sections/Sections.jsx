import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'

import { translate } from 'cozy-ui/react/I18n'
import withBreakpoints from 'cozy-ui/react/helpers/withBreakpoints'

import matcherFromSearch from 'lib/matcherFromSearch'
import {
  getAppsSortedByCategories,
  sortCategoriesAlphabetically,
  getCategoriesSelections
} from 'lib/helpers'

import AppsSection from './components/AppsSection'
import DropdownFilter from './components/DropdownFilter'
import { APP_TYPE } from './constants'
import matches from 'lodash/matches'

const makeOptionMatcherFromSearch = (search = {}) => {
  const typeParam = search.type
  const categoryParam = search.category
  if (typeParam === APP_TYPE.KONNECTOR && !categoryParam) {
    return matches({ value: 'konnectors' })
  } else if (!typeParam && !categoryParam) {
    return matches({ value: 'all' })
  }
  return matches({ value: categoryParam, type: typeParam })
}

const makeSearchFromOption = option => {
  if (option.value === 'all') {
    return {}
  }
  const search = {}
  if (option.value === 'konnectors') {
    search.type = APP_TYPE.KONNECTOR
  } else {
    search.type = option.type
    search.category = option.value
  }
  return search
}

/**
 * Shows a list of apps grouped by categories.
 *
 * Controls an internal search object to filter the list.
 */
export class Sections extends Component {
  constructor(props, context) {
    super(props, context)
    this.state = {
      search: {}
    }
    this.handleSearchOptionChange = this.handleSearchOptionChange.bind(this)
  }

  // Sets state.search from the option received from the DropdownFilter
  handleSearchOptionChange(searchOption) {
    const search = makeSearchFromOption(searchOption)
    if (!this.props.search) {
      // the component is uncontrolled
      this.setState({
        search
      })
    }
    if (typeof this.props.onSearchChange === 'function') {
      this.props.onSearchChange(search)
    }
  }

  render() {
    const { t, apps, error, onAppClick, breakpoints = {}, hasNav } = this.props
    const { isMobile, isTablet } = breakpoints

    if (error) return <p className="u-error">{error.message}</p>

    // Depending on whether the component is controlled or uncontrolled,
    // search is taken from props or state
    const search = this.props.search || this.state.search
    const searchMatcher = matcherFromSearch(search)
    const filteredApps = apps.filter(searchMatcher)

    const konnectorsList = getAppsSortedByCategories(
      filteredApps.filter(a => a.type === APP_TYPE.KONNECTOR)
    )
    const konnectorsCategories = sortCategoriesAlphabetically(
      Object.keys(konnectorsList),
      t
    )

    const webAppsList = getAppsSortedByCategories(
      filteredApps.filter(a => a.type === APP_TYPE.WEBAPP)
    )
    const webAppsCategories = sortCategoriesAlphabetically(
      Object.keys(webAppsList),
      t
    )

    const selectOptions = getCategoriesSelections(apps, t, true)
    const dropdownDisplayed = hasNav && (isMobile || isTablet)

    const optionMatcher = makeOptionMatcherFromSearch(search)
    const defaultFilterValue = selectOptions.find(optionMatcher)

    return (
      <div className={`sto-sections${dropdownDisplayed ? '' : ' u-mt-half'}`}>
        {dropdownDisplayed && (
          <DropdownFilter
            defaultValue={defaultFilterValue}
            options={selectOptions}
            onChange={this.handleSearchOptionChange}
          />
        )}
        {!isMobile && !!webAppsCategories.length && (
          <h1 className="sto-sections-title u-title-h1">
            {t('sections.applications')}
          </h1>
        )}
        <div className="sto-sections-section">
          {!!webAppsCategories.length && (
            <div>
              {webAppsCategories.map(cat => {
                return (
                  <AppsSection
                    key={cat}
                    appsList={webAppsList[cat]}
                    subtitle={
                      <h2 className="sto-sections-subtitle u-title-h2">
                        {t(`app_categories.${cat}`)}
                      </h2>
                    }
                    onAppClick={onAppClick}
                  />
                )
              })}
            </div>
          )}
          {!!konnectorsCategories.length && (
            <div>
              <h2 className="sto-sections-subtitle u-title-h2">
                {t('sections.konnectors')}
              </h2>
              {konnectorsCategories.map(cat => {
                return (
                  <AppsSection
                    key={cat}
                    appsList={konnectorsList[cat]}
                    subtitle={
                      <h3 className="sto-sections-subtitle-secondary">
                        {t(`app_categories.${cat}`)}
                      </h3>
                    }
                    onAppClick={onAppClick}
                  />
                )
              })}
            </div>
          )}
        </div>
      </div>
    )
  }
}

Sections.propTypes = {
  t: PropTypes.func.isRequired,
  apps: PropTypes.array.isRequired,
  error: PropTypes.object,
  onAppClick: PropTypes.func.isRequired,
  hasNav: PropTypes.bool
}

Sections.defaultProps = {
  hasNav: true
}

export default translate()(withBreakpoints()(Sections))
