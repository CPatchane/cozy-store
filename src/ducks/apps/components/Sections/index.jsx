import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'

import { translate } from 'cozy-ui/react/I18n'
import withBreakpoints from 'cozy-ui/react/helpers/withBreakpoints'

import {
  getAppsSortedByCategories,
  sortCategoriesAlphabetically,
  getCategoriesSelections
} from 'lib/helpers'

import AppsSection from './components/AppsSection'
import DropdownFilter from './components/DropdownFilter'
import { APP_TYPE } from './constants'
import isNavigationEnabled from './isNavigationEnabled'
import matches from 'lodash/matches'

const makeOptionMatcherFromSearchParams = params => {
  const typeParam = params.get('type')
  const categoryParam = params.get('category')
  if (typeParam === APP_TYPE.KONNECTOR && !categoryParam) {
    return matches({ value: 'konnectors' })
  } else if (!typeParam && !categoryParam) {
    return matches({ value: 'all' })
  }
  return matches({ value: categoryParam, type: typeParam })
}

export class Sections extends Component {
  handleFilterChange() {
    const { pushQuery } = this.props
    switch (option.value) {
      case 'all':
        return pushQuery()
      case 'konnectors':
        return this.props.pushQuery(`type=${APP_TYPE.KONNECTOR}`)
      default:
        return this.props.pushQuery(`type=${option.type}&category=${option.value}`)
    }
  }

  render() {
    const {
      t,
      apps,
      error,
      onAppClick,
      allApps,
      query,
      pushQuery,
      breakpoints = {},
      location
    } = this.props
    const { isMobile, isTablet } = breakpoints

    if (error) return <p className="u-error">{error.message}</p>

    const konnectorsList = getAppsSortedByCategories(
      apps.filter(a => a.type === APP_TYPE.KONNECTOR)
    )
    const konnectorsCategories = sortCategoriesAlphabetically(
      Object.keys(konnectorsList),
      t
    )

    const webAppsList = getAppsSortedByCategories(
      apps.filter(a => a.type === APP_TYPE.WEBAPP)
    )
    const webAppsCategories = sortCategoriesAlphabetically(
      Object.keys(webAppsList),
      t
    )

    const selectOptions = getCategoriesSelections(allApps, t, true)
    const hasNav = isNavigationEnabled(location.search)
    const dropdownDisplayed = hasNav && (isMobile || isTablet)

    const optionMatcher = makeOptionMatcherFromSearchParams(
      this.state.searchParams
    )
    const defaultFilterValue = selectOptions.find(optionMatcher)

    return (
      <div className={`sto-sections${dropdownDisplayed ? '' : ' u-mt-half'}`}>
        {dropdownDisplayed && (
          <DropdownFilter
            defaultValue={defaultFilterValue}
            options={selectOptions}
            onChange={this.handleFilterChange}
          />
        )}
        {!isMobile && !!webAppsCategories.length && (
          <h1 className="sto-sections-title u-title-h1">
            {t('sections.applications')}
          </h1>
        )}
        <div className="sto-sections-section">
          {!!webAppsCategories.length && (
            <div>
              {webAppsCategories.map(cat => {
                return (
                  <AppsSection
                    key={cat}
                    appsList={webAppsList[cat]}
                    subtitle={
                      <h2 className="sto-sections-subtitle u-title-h2">
                        {t(`app_categories.${cat}`)}
                      </h2>
                    }
                    onAppClick={onAppClick}
                  />
                )
              })}
            </div>
          )}
          {!!konnectorsCategories.length && (
            <div>
              <h2 className="sto-sections-subtitle u-title-h2">
                {t('sections.konnectors')}
              </h2>
              {konnectorsCategories.map(cat => {
                return (
                  <AppsSection
                    key={cat}
                    appsList={konnectorsList[cat]}
                    subtitle={
                      <h3 className="sto-sections-subtitle-secondary">
                        {t(`app_categories.${cat}`)}
                      </h3>
                    }
                    onAppClick={onAppClick}
                  />
                )
              })}
            </div>
          )}
        </div>
      </div>
    )
  }
}

Sections.propTypes = {
  t: PropTypes.func.isRequired,
  apps: PropTypes.array.isRequired,
  error: PropTypes.object,
  onAppClick: PropTypes.func.isRequired,
  allApps: PropTypes.array.isRequired,
  query: PropTypes.string,
  pushQuery: PropTypes.func
}

export default translate()(withBreakpoints()(withRouter(Sections)))
