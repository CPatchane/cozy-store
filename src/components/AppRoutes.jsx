import React from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'

import MyApplications from '../containers/MyApplications'

const AppRoutes = () => {
  return <Switch>
    <Route path='/myapps' component={MyApplications} />
    <Redirect from='/' to='/myapps' />
  </Switch>
}

export default AppRoutes
