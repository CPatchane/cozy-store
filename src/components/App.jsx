
import React from 'react'

import Sidebar from './Sidebar'
import AppRoutes from './AppRoutes'

const App = () => (
  <div className='sto-wrapper coz-sticky'>
    <Sidebar />
    <main className='sto-content'>
      <AppRoutes />
    </main>
  </div>
)
export default App
