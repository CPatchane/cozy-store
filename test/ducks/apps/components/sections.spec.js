'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { tMock } from '../../../jestLib/I18n'
import { Sections } from 'ducks/apps/components/Sections'

import mockApps from '../_mockApps'

Enzyme.configure({ adapter: new Adapter() })

describe('AppsSection component', () => {
  it('should be rendered correctly with apps list, subtitle and onAppClick', () => {
    const mockOnAppClick = jest.fn()
    const component = shallow(
      <Sections
        t={tMock}
        lang="en"
        subtitle="Test Apps"
        apps={mockApps}
        onAppClick={mockOnAppClick}
        error={null}
      />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should render correctly render message if error provided', () => {
    const mockOnAppClick = jest.fn()
    const component = shallow(
      <Sections
        t={tMock}
        lang="en"
        subtitle="Test Apps"
        apps={mockApps}
        onAppClick={mockOnAppClick}
        error={new Error('This is a test error')}
      />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
})
