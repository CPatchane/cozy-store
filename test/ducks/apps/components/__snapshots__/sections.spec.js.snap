// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppsSection component should be rendered correctly with apps list, subtitle and onAppClick 1`] = `
<div
  className="sto-sections"
>
  <h1
    className="sto-sections-title u-title-h1"
  >
    Applications
  </h1>
  <div
    className="sto-sections-section"
  >
    <div>
      <Wrapper
        appsList={
          Array [
            Object {
              "categories": Array [
                "partners",
              ],
              "icon": "<svg></svg>",
              "isInRegistry": true,
              "name": "Tasky",
              "slug": "tasky",
              "type": "webapp",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "1.0.0",
                ],
                "dev": Array [
                  "1.0.0",
                ],
                "stable": Array [
                  "1.0.0",
                ],
              },
            },
          ]
        }
        onAppClick={[MockFunction]}
        subtitle={
          <h2
            className="sto-sections-subtitle u-title-h2"
          >
            Partners
          </h2>
        }
      />
      <Wrapper
        appsList={
          Array [
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Collect",
              "name_prefix": "Cozy",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
                "mock2": Object {
                  "type": "io.mock.doctype2",
                },
              },
              "related": "http://collect.cozy.mock/",
              "slug": "collect",
              "tags": Array [
                "konnector",
                "collect",
                "bills",
                "providers",
                "files",
              ],
              "type": "webapp",
              "uninstallable": false,
              "version": "3.0.0",
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "installed": true,
              "name": "Drive",
              "name_prefix": "Cozy",
              "permissions": Object {
                "mock2": Object {
                  "type": "io.mock.doctype2",
                },
              },
              "related": "http://drive.cozy.mock/",
              "slug": "drive",
              "tags": Array [
                "search",
                "files",
                "folders",
              ],
              "type": "webapp",
              "uninstallable": false,
              "version": "3.0.0-beta89bnhj3993",
            },
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "isInRegistry": true,
              "name": "Photos",
              "name_prefix": "Cozy",
              "slug": "photos",
              "type": "webapp",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                ],
              },
            },
          ]
        }
        onAppClick={[MockFunction]}
        subtitle={
          <h2
            className="sto-sections-subtitle u-title-h2"
          >
            The essentials
          </h2>
        }
      />
      <Wrapper
        appsList={
          Array [
            Object {
              "categories": Array [
                "others",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "isInRegistry": true,
              "locales": Object {
                "name": "For Dev Only",
              },
              "name": "DevOnly",
              "slug": "devonly",
              "type": "webapp",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [],
                "dev": Array [
                  "1.0.0-betaojdkehy989ekhflldh",
                ],
                "stable": Array [],
              },
            },
          ]
        }
        onAppClick={[MockFunction]}
        subtitle={
          <h2
            className="sto-sections-subtitle u-title-h2"
          >
            Others
          </h2>
        }
      />
    </div>
    <div>
      <h2
        className="sto-sections-subtitle u-title-h2"
      >
        Services
      </h2>
      <Wrapper
        appsList={
          Array [
            Object {
              "categories": Array [
                "isp",
                "telecom",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "icon": "<svg></svg>",
              "isInRegistry": true,
              "name": "Bouilligue",
              "slug": "konnector-bouilligue",
              "type": "konnector",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "0.1.0",
                ],
                "dev": Array [
                  "0.1.0",
                ],
                "stable": Array [
                  "0.1.0",
                ],
              },
            },
          ]
        }
        onAppClick={[MockFunction]}
        subtitle={
          <h3
            className="sto-sections-subtitle-secondary"
          >
            ISP
          </h3>
        }
      />
      <Wrapper
        appsList={
          Array [
            Object {
              "categories": Array [
                "isp",
                "telecom",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "icon": "<svg></svg>",
              "isInRegistry": true,
              "name": "Bouilligue",
              "slug": "konnector-bouilligue",
              "type": "konnector",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "0.1.0",
                ],
                "dev": Array [
                  "0.1.0",
                ],
                "stable": Array [
                  "0.1.0",
                ],
              },
            },
          ]
        }
        onAppClick={[MockFunction]}
        subtitle={
          <h3
            className="sto-sections-subtitle-secondary"
          >
            Telecom
          </h3>
        }
      />
      <Wrapper
        appsList={
          Array [
            Object {
              "categories": Array [
                "transport",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Trinlane",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
              },
              "slug": "konnector-trinlane",
              "tags": Array [
                "transport",
                "files",
                "bills",
              ],
              "type": "konnector",
              "uninstallable": true,
              "version": "0.1.0",
              "versions": Object {
                "beta": Array [
                  "0.1.0",
                ],
                "dev": Array [
                  "0.1.0",
                ],
                "stable": Array [
                  "0.1.0",
                ],
              },
            },
          ]
        }
        onAppClick={[MockFunction]}
        subtitle={
          <h3
            className="sto-sections-subtitle-secondary"
          >
            Transportation
          </h3>
        }
      />
    </div>
  </div>
</div>
`;

exports[`AppsSection component should render correctly render message if error provided 1`] = `
<p
  className="u-error"
>
  This is a test error
</p>
`;
